src/binder/bind_node_visitor.cpp

```diff
@@ -440,6 +441,43 @@ void BindNodeVisitor::Visit(common::ManagedPointer<parser::InsertStatement> node
           }
 
           ins_val->Accept(common::ManagedPointer(this).CastManagedPointerTo<SqlNodeVisitor>());
+
+          // if we have constant values of variable types being inserted into fixed size columns,
+          // we need to resolve that
+          if (ins_val->GetExpressionType() == parser::ExpressionType::VALUE_CONSTANT) {
+            auto const_val = ins_val.CastManagedPointerTo<parser::ConstantValueExpression>();
+            bool is_variable = const_val->GetReturnValueType() == type::TypeId::VARCHAR ||
+                               const_val->GetReturnValueType() == type::TypeId::VARBINARY;
+            if (is_variable && !const_val->IsNull() && ins_col.TypeModifier() != -1 &&
+                const_val->GetStringVal().GetLength() > static_cast<size_t>(ins_col.TypeModifier())) {
+              // postgres truncates off trailing spaces from the end of the varchar only to the max size of the col
+              // if it is still too large it throws an error
+              // https://github.com/postgres/postgres/blob/master/src/backend/utils/adt/varchar.c
+              // issue #1380
+              size_t num_trailing_spaces = 0;
+              for (size_t idx = const_val->GetStringVal().GetLength() - 1;
+                   idx < const_val->GetStringVal().GetLength() && idx >= static_cast<size_t>(ins_col.TypeModifier());
+                   idx--) {
+                if (const_val->GetStringVal().GetContent()[idx] == ' ')
+                  num_trailing_spaces++;
+                else
+                  break;
+              }
+              size_t true_len = const_val->GetStringVal().GetLength() - num_trailing_spaces;
```

Comment:
`true_length` is fine. We try not to abbreviate variable names much.

Reply:
done